### PR TITLE
[Finder] simplified code

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php [hunk 1]
```diff
@@ -87,7 +85,20 @@ protected function isAccepted($string)
...
-        return Expression::create($str)->isRegex();
+        if (preg_match('/^(.{3,}?)[imsxuADU]*$/', $str, $m)) {
+            $start = substr($m[1], 0, 1);
+            $end = substr($m[1], -1);
+
+            if ($start === $end) {
+                return !preg_match('/[*?[:alnum:] \\\\]/', $start);
+            }
+
+            if ($start === '{' && $end === '}') {
```
Comment (stof) [inline src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php]: these are not the only bracket types supported for delimiters btw. Our code is missing support for other ones: http://php.net/manual/en/regexp.reference.delimiters.php

Comment (fabpot) [inline src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php]: fixed

Comment (stof) [inline src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php]: btw, the logic in the Expression subsystem was supporting ``()`` as delimiters, so this is actually a regression

**Comment by stof** [comment]
:+1: this complex system is not necessary anymore now that we deprecated the find-based adapters.

@fabpot is it possible to add tests for the regex detection here, including the case of all kinds of bracket delimiters ?
The logic was currently covered by tests of the Expression class, but we need to cover it by itself now as we won't call the Expression class anymore here.

and you need to mark tests of the ``Expression`` classes as legacy ones to avoid issues with deprecation warnings.

**Comment by fabpot** [comment]
The tests where already in the iterator test class for `isRegex()`, but I've just added some more.
We should be good now.

**Comment by stof** [comment]
good to me

**Pull request merged**